packages/core/src/ruleset/meta/shared.json

```diff
@@ -50,9 +50,26 @@
     },
     "PathExpression": {
       "$id": "path-expression",
-      "type": "string",
-      "pattern": "^[$#]",
-      "errorMessage": "must be a valid JSON Path expression or a reference to the existing Alias optionally paired with a JSON Path expression subset"
+      "if": {
+        "type": "string"
+      },
+      "then": {
```

Comment:
This conditional logic stuff in json schema is cool and terrifying at the same time.